internal/zinc-classpath/src/main/scala/sbt/internal/inc/classpath/ClasspathUtilities.scala

```diff
@@ -63,8 +63,44 @@ object ClasspathUtilities {
   final val AppClassPath = "app.class.path"
   final val BootClassPath = "boot.class.path"
 
-  def createClasspathResources(classpath: Seq[File], instance: ScalaInstance): Map[String, String] =
-    createClasspathResources(classpath, instance.allJars)
+  private[sbt] def findLibrary(instance: ScalaInstance): File = {
+    instance match {
+      case i: inc.ScalaInstance => i.libraryJar
+      case _ =>
+        (instance.allJars find { x =>
+          x.getName.startsWith("scala-library")
+        }).getOrElse {
+          throw new InvalidScalaProvider(s"Couldn't find scala-library")
+        }
+    }
+  }
+
+  /** returns the cached library loader if it's inc.ScalaInstance. */
+  def getLibraryLoader(instance: ScalaInstance): ClassLoader = {
+    instance match {
+      case i: inc.ScalaInstance => i.libraryLoader
+      case _                    => toLibraryLoader(instance)
+    }
+  }
+
+  /**
+   * Include only the scala-library.jar, if any, into the boot classpath.
+   * 1. Supposedly using boot classpath here is for performance reason:
+   *    https://groups.google.com/d/msg/scala-internals/tT1pjH5GECE/dtgRj3w7ovIJ
```

Comment:
"boot classpath" isn't an accurate description of what's happening here. The performance trick is to cache and share `libraryLoader` in `ScalaInstance`, rather than forming new classpaths containing `scala-library.jar`.

Reply:
`createClasspathResources(cp, instance)` does map things into `BootClassPath`.